### PR TITLE
2169 pr-bot should ignore case in "Backport <sha>" PR titles

### PR DESCRIPTION
**Pull request by LudwikJaniuk**
There's no reason why this should validate case, and has caused unnecessary friction in the past.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-2169](https://bugs.openjdk.org/browse/SKARA-2169): pr-bot should ignore case in "Backport &lt;sha&gt;" PR titles (**Enhancement** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1611/head:pull/1611` \
`$ git checkout pull/1611`

Update a local copy of the PR: \
`$ git checkout pull/1611` \
`$ git pull https://git.openjdk.org/skara.git pull/1611/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1611`

View PR using the GUI difftool: \
`$ git pr show -t 1611`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1611.diff">https://git.openjdk.org/skara/pull/1611.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1611#issuecomment-1934187734)

### CODE REVIEW

**Comment by erikj79** [comment]
Please file a bug for this and associate with the PR. You can also enable "actions" for your fork to automatically run the tests.

**Comment by LudwikJaniuk** [comment]
`sh gradlew build` passes with the changes.

**Comment by lgxbslgx** [comment]
It is good to add some test cases to verify this changement.

File: bots/pr/src/test/java/org/openjdk/skara/bots/pr/BackportTests.java [hunk 1]
```diff
@@ -1137,6 +1137,65 @@ void whitespaceAtEnd(TestInfo testInfo) throws IOException {
...
+            // "backport" the new file to the master branch
+            localRepo.checkout(localRepo.defaultBranch());
+            var editBranch = localRepo.branch(masterHash, "edit");
+            localRepo.checkout(editBranch);
+            var newFile2 = localRepo.root().resolve("a_new_file.txt");
+            Files.writeString(newFile2, "hello");
+            localRepo.add(newFile2);
+            var editHash = localRepo.commit("Backport", "duke", "duke@openjdk.org");
+            localRepo.push(editHash, author.authenticatedUrl(), "refs/heads/edit", true);
+            var pr = credentials.createPullRequest(author, "master", "edit", "bacKporT" + releaseHash.hex());
```
Comment (erikj79) [inline bots/pr/src/test/java/org/openjdk/skara/bots/pr/BackportTests.java:1187]: I hadn't realized this, but we actually don't require space between `backport` and the hash. That's probably fine and not something we need to change here.

**Review by erikj79 — APPROVED** [review]

File: bots/pr/src/test/java/org/openjdk/skara/bots/pr/BackportTests.java (hunk 1 above)
Comment (LudwikJaniuk) [reply bots/pr/src/test/java/org/openjdk/skara/bots/pr/BackportTests.java:1187]: Yep

**Pull request closed without merging**